crates/oxc_query/src/edges.rs

```diff
@@ -0,0 +1,2049 @@
+        _resolve_info: &ResolveEdgeInfo,
+        adapter: &'a Adapter<'b>,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            let i = *v.as_path_part().unwrap();
+            if i + 1 < adapter.path_components.len() {
+                Box::new(std::iter::once(Vertex::PathPart(i + 1)))
+            } else {
+                Box::new(std::iter::empty())
+            }
+        })
+    }
+
+    pub(super) fn prev<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            let i = *v.as_path_part().unwrap();
+            if i > 0 {
+                Box::new(std::iter::once(Vertex::PathPart(i - 1)))
+            } else {
+                Box::new(std::iter::empty())
+            }
+        })
+    }
+}
+
+pub(super) fn resolve_return_statement_ast_edge<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    edge_name: &str,
+    _parameters: &EdgeParameters,
+    resolve_info: &ResolveEdgeInfo,
+    adapter: &'a Adapter<'b>,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+    match edge_name {
+        "ancestor" => return_statement_ast::ancestor(contexts, resolve_info, adapter),
+        "expression" => return_statement_ast::expression(contexts, resolve_info),
+        "parent" => return_statement_ast::parent(contexts, resolve_info, adapter),
+        "span" => return_statement_ast::span(contexts, resolve_info),
+        _ => {
+            unreachable!(
+                "attempted to resolve unexpected edge '{edge_name}' on type 'ReturnStatementAST'"
+            )
+        }
+    }
+}
+
+mod return_statement_ast {
+    use trustfall::provider::{
+        resolve_neighbors_with, ContextIterator, ContextOutcomeIterator, ResolveEdgeInfo,
+        VertexIterator,
+    };
+
+    use super::super::vertex::Vertex;
+    use crate::Adapter;
+
+    pub(super) fn ancestor<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+        adapter: &'a Adapter<'b>,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        super::ancestors(contexts, adapter)
+    }
+
+    pub(super) fn expression<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            #[allow(clippy::option_if_let_else)]
+            if let Some(expr) = &v.as_return_statement_ast().unwrap().return_statement.argument {
+                Box::new(std::iter::once(expr.into()))
+            } else {
+                Box::new(std::iter::empty())
+            }
```

Comment:
you should just `.into_iter()` on the `Option` directly, it does the correct thing and clippy is right to complain

Reply:
That's not why clippy complains, it's because of this: https://github.com/rust-lang/rust-clippy/issues/11059

Reply:
I also don't really understand how you would fix this with .into_iter, more explanation would be amazing

Reply:
Amazing, thanks! done in [275986e](https://github.com/web-infra-dev/oxc/pull/599/commits/275986ef1119b9f86c3de211f13fa39a80b555b4)

Reply:
I think you used `.unwrap()` again in that commit, but maybe you've since fixed it.

Reply:
Fixed! now it's into_iter()